### PR TITLE
More dumping stuff

### PR DESCRIPTION
**Pull request by Cubxity**
so far everything else is fine but the modifier "/* synthetic */" and such still prints as keyword

### CODE REVIEW

**Comment by leibnitz27** [comment]
Hi - did you run this through cfr-test?

It fails on the first check. 

Checking AbeDescent
1,3c1,7
< /*
<  * Decompiled with CFR.
<  */
---
> // /*
>
> //  *
> // Decompiled with CFR.
>
> //  */

**Comment by Cubxity** [comment]
Hi, I haven't ran the code through cfr-test. I'll look into the problem

**Comment by leibnitz27** [comment]
This will be because block comments aren't handled specially.   In order to do this, the dumper will need a beginBlockComment and endBlockComment.    If you fancy having a go at this, feel free, otherwise I'll add it.

**Comment by Cubxity** [comment]
@leibnitz27 The comments are looking fine now in my tool, would you mind running the test for me since I don't have all the java version required installed?

image

**Comment by leibnitz27** [comment]
Will do. ;)

**Comment by leibnitz27** [comment]
Hey - this is still adding comment markers infront of block comments.

You don't need all the versions of java installed to run the tests, but it will really speed merging if you run at least one of them.   

I'll introduce some helpers for block comments and get back to you.

Checking AbeDescent
1,3c1,7
< /*
<  * Decompiled with CFR.
<  */
---
> // /*
>
> //  *
> // Decompiled with CFR.
>
> //  */
>

**Comment by Cubxity** [comment]
Alright

File: src/org/benf/cfr/reader/entities/classfilehelpers/AbstractClassFileDumper.java [hunk 1]
```diff
@@ -68,28 +68,28 @@ void dumpTopHeader(ClassFile classFile, Dumper d) {
...
         if (formalTypeParameters == null || formalTypeParameters.isEmpty()) return;
-        d.print('<');
+        d.separator("<");
         boolean first = true;
         for (FormalTypeParameter formalTypeParameter : formalTypeParameters) {
             first = StringUtils.comma(first, d);
             d.dump(formalTypeParameter);
         }
-        d.print('>');
+        d.print(">");
```
Comment (Marcono1234) [inline src/org/benf/cfr/reader/entities/classfilehelpers/AbstractClassFileDumper.java]: Shouldn't this be a separator as well?

File: src/org/benf/cfr/reader/entities/classfilehelpers/ClassFileDumperAnnotation.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ public Dumper dump(ClassFile classFile, InnerClassDumpType innerClass, Dumper d)
         boolean first = true;
         dumpAnnotations(classFile, d);
         dumpHeader(classFile, innerClass, d);
-        d.print("{").newln();
+        d.separator("{").newln();
```
Comment (Marcono1234) [inline src/org/benf/cfr/reader/entities/classfilehelpers/ClassFileDumperAnnotation.java]: Should the closing `}` in line 67 be a separator as well?

Same for ClassFileDumperAnonymousInner.java, ClassFileDumperEnum.java and ClassFileDumperInterface.java.

**Pull request merged**